Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -326,31 +334,51 @@ else if (e.getSeverity() == CqlTranslatorException.ErrorSeverity.Info) {
         }
     }
 
+    private boolean shouldReport(CqlTranslatorException.ErrorSeverity errorSeverity) {
+        switch (errorLevel) {
+            case Info:
+                return
+                        errorSeverity == CqlTranslatorException.ErrorSeverity.Info
+                                || errorSeverity == CqlTranslatorException.ErrorSeverity.Warning
+                                || errorSeverity == CqlTranslatorException.ErrorSeverity.Error;
+            case Warning:
+                return
+                        errorSeverity == CqlTranslatorException.ErrorSeverity.Warning
+                                || errorSeverity == CqlTranslatorException.ErrorSeverity.Error;
+            case Error:
+                return errorSeverity == CqlTranslatorException.ErrorSeverity.Error;
+            default:
+                throw new IllegalArgumentException(String.format("Unknown error severity %s", errorSeverity.toString()));
+        }
+    }
```

Comment:
If by implementation detail, you mean the order in which the enum is defined, then that is true.  And that is likely why some might consider it bad practice to use it.  I suppose if you did use it, it would warrant a comment in the enum definition to indicate that order DOES matter.

It wouldn't be difficult to update the enum to carry a _level_ int.  Something like:
```java
public enum ErrorSeverity {
    Info(0),
    Warning(1),
    Error(2)

    private int level;

    private ErrorSeverity(int level) {
        this.level = level;
    }

    public int getLevel() { return level; }
}
```
Then you could use `getLevel()` instead of `ordinal()` (BTW -- I didn't try to compile that, so may not be perfect).

All that said, I won't fail the pull request on this -- so if you prefer to keep it as-is, that's OK.